multiapi-engine/src/main/java/com/sngular/api/generator/plugin/openapi/utils/MapperPathUtil.java

```diff
@@ -47,12 +47,13 @@ private MapperPathUtil() {}
 
   public static GlobalObject mapOpenApiObjectToOurModels(final OpenAPI openAPI, final List<AuthSchemaObject> authSchemaList) {
     final var authList = getSecurityRequirementList(openAPI.getSecurity(), new ArrayList<>());
-    return GlobalObject.builder()
-                       .url(openAPI.getServers().get(0).getUrl())
-                       .authSchemas(authSchemaList)
-                       .authentications(authList)
-                       .schemaMap(openAPI.getComponents().getSchemas())
-                       .build();
+    final GlobalObject globalObject = GlobalObject.builder().url(openAPI.getServers().get(0).getUrl()).authSchemas(authSchemaList).authentications(authList).build();
+    if (Objects.nonNull(openAPI.getComponents())) {
+      globalObject.setSchemaMap(openAPI.getComponents().getSchemas());
+    } else {
+      globalObject.setSchemaMap(Collections.emptyMap());
+    }
+    return globalObject;
```

Comment:
Since this is a builder, a more appropiate way of doing this would be:
```suggestion
    final GlobalObject globalObject = GlobalObject.builder().url(openAPI.getServers().get(0).getUrl()).authSchemas(authSchemaList).authentications(authList);
    globalObject.schemaMap(Objects.nonNull(openAPI.getComponents()) ? openAPI.getComponents().getSchemas() : Collections.emptyMap());
    return globalObject.build();
```